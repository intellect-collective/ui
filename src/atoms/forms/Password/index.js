import React from 'react';
import changeable from '../_decorators/changeable';
import formContext from '../_decorators/formContext';

const Password = (props) => (
    <input { ...props } type="password" />
);
Password.displayName = 'Password';

export default formContext(changeable(Password));
