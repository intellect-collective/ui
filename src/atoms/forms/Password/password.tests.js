import React from 'react';
import Password from '.';
import {
    basicChangeableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    value: 'secret',
    onClick: () => {}
};
const expectedProps = {
    name: 'test',
    value: 'secret',
    onClick: expect.any(Function)
};

describe('Password', () => {
    basicChangeableTests(Password, 'input', validProps, expectedProps);
});
