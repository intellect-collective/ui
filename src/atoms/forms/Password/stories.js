import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Password from './index';

storiesOf('Atoms / Forms / Password', module)
    .add('default', () => (<Password name="test-pass-1" onChange={ action('onChange') } />));
