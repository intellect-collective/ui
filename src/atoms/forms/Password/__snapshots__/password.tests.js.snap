// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Should accept valid props 1`] = `
<FormContext
  name="test"
  onClick={[Function]}
  value="secret"
>
  <Changeable
    name="test"
    onClick={[Function]}
    value="secret"
  >
    <Password
      name="test"
      onChange={[Function]}
      onClick={[Function]}
      value="secret"
    >
      <input
        name="test"
        onChange={[Function]}
        onClick={[Function]}
        type="password"
        value="secret"
      />
    </Password>
  </Changeable>
</FormContext>
`;
